src/basefoc.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * Source code for Urban motor controller functionality
+ */
+
+#include "../include/main.h"
+extern Encoder encoder;
+
+//  BLDCMotor(int pp, (optional R, KV))
+//  - pp  - pole pair number
+//  - R   - phase resistance value - optional
+//  - KV  - motor KV rating [rpm/V] - optional
+// For Surpass C2216-880KV motor | 14 pole pairs | 0.108 Resistance | 880KV
+BLDCMotor motor = BLDCMotor(14);
+
+// For Koford 129H169T motor | 10 pole pairs | TBD Resistance | TBD
+// BLDCMotor motor = BLDCMotor(10);
+
+//  BLDCDriver3PWM( int phA, int phB, int phC, int enA, int enB, int enC )
+//  - phA, phB, phC - A,B,C phase pwm pins
+//  - enA, enB, enC - enable pin for each phase (optional)
+BLDCDriver3PWM driver = BLDCDriver3PWM(12, 13, 14);
+
+// InlineCurrentSensor constructor
+//  - shunt_resistor  - shunt resistor value
+//  - gain  - current-sense op-amp gain
+//  - phA   - A phase adc pin
+//  - phB   - B phase adc pin
+//  - phC   - C phase adc pin (optional - _NC if not using)
+InlineCurrentSense current_sense  = InlineCurrentSense(0.01, 50.0, A0, T1, T2);
+
+// commander communication instance
+Commander command = Commander(Serial);
+void onMotor(char* cmd){ command.motor(&motor, cmd); }
+
+void BaseFOC( void * pvParameters ) {
+
+  Serial.println("Base Ready");
+  motor.useMonitoring(Serial);
+
+  // link the motor to the sensor
+  motor.linkSensor(&encoder);
+
+  // power supply voltage [V] - Change this locally for testing
+  driver.voltage_power_supply = 48;
+  // Max DC voltage allowed - default voltage_power_supply
+  driver.voltage_limit = 50.4;
+  driver.pwm_frequency = 30000; //based on mcu used
+  driver.init();
+  // link driver
+  motor.linkDriver(&driver);
+  // link current sense and driver
+  current_sense.linkDriver(&driver);
+
+  // aligning voltage
+  motor.voltage_sensor_align = 3; // Volts
+
+  // choose FOC modulation
+  motor.foc_modulation = FOCModulationType::SinePWM;
+
+  // set motion control loop to be used
+  motor.controller = MotionControlType::velocity;
+
+  // setting the limits - Change this locally for testing
+  // either voltage
+  // motor.voltage_limit = 50.4; // Volts - default driver.voltage_limit
+  // or current 
+  motor.current_limit = 40; // Amps - default 0.2Amps
+
+  // angle loop velocity limit
+  motor.velocity_limit = 50;
+
+  // monitoring values to display
+  // motor.monitor_variables = _MON_TARGET | _MON_VEL | _MON_ANGLE; // default _MON_TARGET | _MON_VOLT_Q | _MON_VEL | _MON_ANGLE
+
+  // initialise motor
+  motor.init();
+
+  // initialise the current sensing
+  current_sense.init();
```

Comment:
I don't know if it is a good idea to initialize the current sensing in this program since we are only doing velocity control most likely (as Luke said) this year, but only velocity control.

Reply:
I will keep it. For now, it can be locally commented in and out, as will be done with other parameters of the setup.